### DISCUSSION TITLE
Skill based matchmaking

### DISCUSSION THREAD

**Discussion by squivix** [TheSuperHackers/GeneralsGamePatch#2483] [Ideas]
Hello. I'd like to suggest a skill-based matchmaking and ranking system. This makes competitive games more fun for everyone (prevents noobs feeling discouraged and pros feeling not challenged). For an example from a different genre look at [Fightcade](https://www.patreon.com/posts/how-fightcades-42104017)

Thank you.

**Comment by MarcosV2020** [comment]
Yeah, i was a fightcade semi pro back in the days. I know what you mean. The ranks are more polished and clean, which means you can enter a Rank A room for example easily than trying to find a room called by a description "pro room". I totally agree with you and i think many pros will like this option also. A nice idea to give more organization and little detail improvement.

**Comment by ghost** [comment]
man, I would love to see the Online Matchmaking come back to life.

**Comment by RetroMous3** [comment]
For some push back! :)
Is there really a need for it?! 

With rooms like when a host opens up a game say  1v1 with the option of “noob rules,” “pro rules,” or “no rush 10,” ETC being selected... plus the edition of the request of "noobs only" or "pros only" (in the game heading like on gameranger) for that match up, players and the community have been and can already self-sort by skill and preference. A 1v1 “pro rules” or "NO rules" room naturally attracts higher-skill players, while “noob rules” gives beginners a sandbox. that also being said that a FFA or pro rules etc, pros and noobs already know what they are getting themselves into. the noob might very well know its a tall order and doesnt expect to win but plays any way because this type of guy likes playing people better than him. this geezer is also going to see some high level playing he would never experience playing a fellow noob..

Although the Fightcade idea works there over at fightcade for a game like Xmen VS Street Fighter,  there are many reasons why it might very well not end up working for ZeroHour (RTS) or even the feature being left redundant

It could be added to a QuickMatch feature that will match make by skill level and ranking system but A “quick match” is a blind, fast toss-in makes more sense for players who dont care about rank  and it doesn’t need ranks..its about hurry up who cares lets play .  its more for casual players who just want to blaze into a game and not  mess around with rooms and game modes.

Add in team modes (2v2, 4v4) or quirky setups , and you’ve got variety without needing an algorithm. If “quick match” is just a fast-track button for folks too busy to browse, it doesn’t cry out for Skill based matchmaking  it’s about convenience, not a skill solution.


the idea might also tank the queue times if the player base is small. imagine waiting forever for a “Rank C” match because there aren’t enough mid-tier players online.. and also on top of and compounded by the issue of "PING" plus region times etc there will already be few games to select from. adding walls to players might also discourage people from opening ZeroHour at all.

If players then have to  face the grim reality of a few rooms due to PING that are playable for them, and then now also have to be qualified to enter a room due to segregation like a NFS like A class, B class, C class  type framework and hope that players of that skill set are online to play, imo isnt needed and wont benefit the community as a whole 

And to add that last little push.... if noobs are always fighting noobs they will always be noobs
